src/Compilers/CSharp/Test/Emit3/PartialEventsAndConstructorsTests.cs

```diff
@@ -1411,4 +1430,625 @@ void M()
+                internal partial event System.Action G { add { } remove { } }
+            }
+            """;
+        CreateCompilation(source).VerifyDiagnostics(
+            // (10,20): error CS8799: Both partial member declarations must have identical accessibility modifiers.
+            //     public partial C() { }
+            Diagnostic(ErrorCode.ERR_PartialMemberAccessibilityDifference, "C").WithLocation(10, 20),
+            // (11,21): error CS8799: Both partial member declarations must have identical accessibility modifiers.
+            //     private partial C(int x) { }
+            Diagnostic(ErrorCode.ERR_PartialMemberAccessibilityDifference, "C").WithLocation(11, 21),
+            // (12,41): error CS8799: Both partial member declarations must have identical accessibility modifiers.
+            //     private partial event System.Action E { add { } remove { } }
+            Diagnostic(ErrorCode.ERR_PartialMemberAccessibilityDifference, "E").WithLocation(12, 41),
+            // (14,42): error CS8799: Both partial member declarations must have identical accessibility modifiers.
+            //     internal partial event System.Action G { add { } remove { } }
+            Diagnostic(ErrorCode.ERR_PartialMemberAccessibilityDifference, "G").WithLocation(14, 42));
+    }
+
+    [Fact]
+    public void Difference_Type()
+    {
+        var source = """
+            using A = System.Action;
+            partial class C
+            {
+                partial event System.Action E, F;
+                partial event System.Action<(int X, int Y)> G;
+                partial event System.Action<(string X, string Y)> H;
+                partial event System.Action<dynamic> I;
+                partial event A J;
+            }
+            partial class C
+            {
+                partial event System.Func<int> E { add { } remove { } }
+                partial event System.Action F { add { } remove { } }
+                partial event System.Action<(int A, int B)> G { add { } remove { } }
+                partial event System.Action<(int A, int B)> H { add { } remove { } }
+                partial event System.Action<object> I { add { } remove { } }
+                partial event System.Action J { add { } remove { } }
+            }
+            """;
+        CreateCompilation(source).VerifyDiagnostics(
+            // (12,36): error CS9255: Both partial member declarations must have the same type.
+            //     partial event System.Func<int> E { add { } remove { } }
+            Diagnostic(ErrorCode.ERR_PartialMemberTypeDifference, "E").WithLocation(12, 36),
+            // (14,49): error CS8142: Both partial member declarations, 'C.G' and 'C.G', must use the same tuple element names.
+            //     partial event System.Action<(int A, int B)> G { add { } remove { } }
+            Diagnostic(ErrorCode.ERR_PartialMemberInconsistentTupleNames, "G").WithArguments("C.G", "C.G").WithLocation(14, 49),
```

Comment:
>ERR_PartialMemberInconsistentTupleNames [](http://example.com/codeflow?start=33&length=39)

Are we reporting this error for partial constructor parameters?

Reply:
Never mind, it looks like that case results in `WRN_PartialMemberSignatureDifference` instead, in test below.